consensus/istanbul/backend/api.go

```diff
@@ -66,39 +66,42 @@ func (api *API) GetSnapshotAtHash(hash common.Hash) (*Snapshot, error) {
 	return api.istanbul.snapshot(api.chain, header.Number.Uint64(), header.Hash(), nil)
 }
 
-// GetValidators retrieves the list of authorized validators at the specified block.
+// GetValidators retrieves the list of authorized validators with the given block number.
 func (api *API) GetValidators(number *rpc.BlockNumber) ([]common.Address, error) {
-	// Retrieve the requested block number (or current if none requested)
-	var header *types.Header
-	if number == nil || *number == rpc.LatestBlockNumber {
-		header = api.chain.CurrentHeader()
-	} else {
-		header = api.chain.GetHeaderByNumber(uint64(number.Int64()))
-	}
-	// Ensure we have an actually valid block and return the validators from its snapshot
-	if header == nil {
-		return nil, errUnknownBlock
-	}
-	snap, err := api.istanbul.snapshot(api.chain, header.Number.Uint64(), header.Hash(), nil)
+	snap, err := api.GetSnapshot(number)
 	if err != nil {
 		return nil, err
 	}
 	return snap.validators(), nil
 }
 
-// GetValidatorsAtHash retrieves the state snapshot at a given block.
+// GetValidatorsAtHash retrieves the list of authorized validators with the given block hash.
 func (api *API) GetValidatorsAtHash(hash common.Hash) ([]common.Address, error) {
-	header := api.chain.GetHeaderByHash(hash)
-	if header == nil {
-		return nil, errUnknownBlock
-	}
-	snap, err := api.istanbul.snapshot(api.chain, header.Number.Uint64(), header.Hash(), nil)
+	snap, err := api.GetSnapshotAtHash(hash)
 	if err != nil {
 		return nil, err
 	}
 	return snap.validators(), nil
 }
 
+// GetDemotedValidators retrieves the list of authorized, but demoted validators with the given block number.
+func (api *API) GetDemotedValidators(number *rpc.BlockNumber) ([]common.Address, error) {
+	snap, err := api.GetSnapshot(number)
```

Comment:
Is it okay to call api in api?
I'm curious about other people's opinions.

Reply:
I think it's okay.

Reply:
I think it makes it simpler and less complex and it is fine logically as well. If you have any other suggestion, I appreciate for your opinion.

Reply:
This is just my opinion that it is awkward for API to call API.
I don't want it to sound rude.
1. API is code written for external communication. It feels awkward to be called by code from inside.
2. The GetDemotedValidators API will be affected if modifications or deletions are made to the GetSnapshot API. I think this will be a less flexible structure.
3. So If there is a duplicate code between multiple APIs, I think it is appropriate to separate that part into other method.

Reply:
@hyochan-brown Thanks for your opinion. According to the other cases, I think this would be fine. There are already many cases for the API structure used its method. I think it makes more consistent with other APIs. What do you think about this?

`s.DoEstimateGas`
`s.ChainId()`
`s.SendTransaction`
`s.SendTransactionAsFeePayer`
`s.SignAndSendTransaction`
...

Reply:
Oh..
Consistency is more important than anything else, so I think it's good. 
Thank you for your kind reply.